src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,34 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    static final byte dateIndex = 0;
+    static final byte nameIndex = 1;
+    static final byte workingDaysIndex = 2;
+    static final byte hourlyRateIndex = 3;
+    static final DateTimeFormatter formatter =
+            DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder report = new StringBuilder("Report for period " + dateFrom
+                + " - " + dateTo);
+        LocalDate dtFrom = LocalDate.parse(dateFrom, formatter).minusDays(1);
+        LocalDate dtTo = LocalDate.parse(dateTo, formatter).plusDays(1);
+        int[] salary = new int[names.length];
+        for (int i = 0; i < names.length; i++) {
+            for (String z:data) {
+                String[] dataArray = z.split(" ");
+                LocalDate neededDt = LocalDate.parse(dataArray[dateIndex], formatter);
+                if (neededDt.isAfter(dtFrom) && neededDt.isBefore(dtTo)
+                        && names[i].equals(dataArray[nameIndex])) {
+                    salary[i] += Integer.parseInt(dataArray[workingDaysIndex])
+                            * Integer.parseInt(dataArray[hourlyRateIndex]);
+                }
+            }
+            report.append(System.lineSeparator() + names[i] + " - " + salary[i]);
```

Comment:
```suggestion
            report.append(System.lineSeparator()).append(names[i]).append(" - ").append(salary[i]);
```

Reply:
experiments with concatenation inside a StringBuilder shows that the code runs faster than with each time .append.